### DISCUSSION TITLE
Problem of photonic labelling

### DISCUSSION THREAD

**Discussion by CQUswl** [nrc-cnrc/EGSnrc#1087] [General]
How can photons passing through a particular scoring surface be labelled?
For example, for photons produced by an electron beam impacting a 1mm tungsten target, I would only want to record photons intersecting a 45 x 45cm size field at 85cm.

**Comment by rtownson** [comment]
In BEAMnrc you can use LATCH in order to label particles as they pass through or interact within selected regions. If you can create the geometries in BEAMnrc, this might be the way to go. LATCH is not yet implemented in egs++. However, you could write an egs++ application that does this exactly according to your needs.

I'm not entirely sure what you're actually requiring, so it may be simpler than this. If you just want to record photons photons within that square into a phase-space, you can do this in egs++ using egs_phsp_scoring.

**Comment by CQUswl** [comment]
> In BEAMnrc you can use LATCH in order to label particles as they pass through or interact within selected regions. If you can create the geometries in BEAMnrc, this might be the way to go. LATCH is not yet implemented in egs++. However, you could write an egs++ application that does this exactly according to your needs.
> 
> I'm not entirely sure what you're actually requiring, so it may be simpler than this. If you just want to record photons photons within that square into a phase-space, you can do this in egs++ using egs_phsp_scoring.

I ran the simulation on BEAMnrc_GUI.The specific problem is that I simulated an electron beam hitting a target.A phase space file was obtained after the target(Z=1 mm).I would like to read the photons in this phase space file that point to a specific plane, i.e. a field of size 45 x 45 cm2 at 85 cm.I don't know if BEAMnrc can do this operation.

**Reply by rtownson** [reply]
So it sounds like you just want to project the photons through a vacuum and collect another phase-space at 85cm. You could do this with a BEAMnrc simulation in vacuum where it uses your existing phase-space as a source, a big slab of vacuum (45cm wide), and scoring a new phase-space after the slab.

**Reply by CQUswl** [reply]
I am not trying to collect phase space file at 85 cm. I just want to know which photons in the phase space file at 0 cm reach a particular recording surface at 85 cm. What I care about is the phase space file at 0 cm.

**Reply by rtownson** [reply]
I think you'll have to write a little custom code to do this. You can use the existing tools in EGSnrc to read the phase-space, get a particle, and then project it to 85cm. Then do whatever you want with those particles. You could write a standalone code to do this, or even program it into a little egs++ app. Either way will take some programming skills. If you do attempt it, you're welcome to ask for help.

**Comment by CQUswl** [comment]
> I think you'll have to write a little custom code to do this. You can use the existing tools in EGSnrc to read the phase-space, get a particle, and then project it to 85cm. Then do whatever you want with those particles. You could write a standalone code to do this, or even program it into a little egs++ app. Either way will take some programming skills. If you do attempt it, you're welcome to ask for help.

I probably understand what it means. Which existing tool in egsnrc will fulfil my requirements?

**Reply by rtownson** [reply]
There is not one, you will need to create one for yourself.
